lib/config.js

```diff
@@ -199,6 +199,21 @@ var normalizeConfig = function (config, configFilePath) {
     throw new TypeError('Invalid configuration: formatError option must be a function.')
   }
 
+  // if the user changed listenAddress, but didn't set a hostname, warn them
+  var testConfig = new Config()
+  testConfig.hostname = null
+  testConfig.listenAddress = null
+  if (configModule != null) {
+    configModule(testConfig)
```

Comment:
Let's not do this, as this means we are evaluating the code from the user twice. Best case this slows things down, worst case it has unexpected side effects. You can add the check probably around the original invocition of the `configModule`